src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.Map.cs

```diff
@@ -5,16 +5,28 @@
 #nullable enable
 using System.Collections.Generic;
 using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
-using System.Threading;
 
 namespace System.Formats.Cbor
 {
     public partial class CborReader
     {
         private KeyEncodingComparer? _keyEncodingComparer;
-        private Stack<HashSet<(int Offset, int Length)>>? _pooledKeyEncodingRangeSets;
-
+        private Stack<HashSet<(int Offset, int Length)>>? _pooledKeyEncodingRangeAllocations;
+
+        /// <summary>
+        ///   Reads the next data item as the start of a map (major type 5)
+        /// </summary>
+        /// <returns>
+        ///   The length of the definite-length map, or <c>null</c> if the array is indefinite-length.
+        /// </returns>
+        /// <exception cref="InvalidOperationException">
+        ///   the next data item does not have the correct major type.
+        /// </exception>
+        /// <exception cref="FormatException">
+        ///   invalid CBOR encoding data --OR--
+        ///   unexpected end of CBOR encoding data --OR--
+        ///   CBOR encoding not accepted under the current conformance level
+        /// </exception>
         public int? ReadStartMap()
```

Comment:
This method probably warrants a remarks section about how the return value represents pairs, and that maps just read like arrays with twice as many elements as the reported size here, key0, value0, key1, value1, etc.  But in words that don't already presuppose the reader knows that :smile:.